File: examples/leetcode-out/hs/25/reverse-nodes-in-k-group.hs
module Main where

import Data.Maybe (fromMaybe)


forLoop :: Int -> Int -> (Int -> Maybe a) -> Maybe a
forLoop start end f = go start
  where
    go i | i < end =
            case f i of
              Just v -> Just v
              Nothing -> go (i + 1)
         | otherwise = Nothing

avg :: Real a => [a] -> Double
avg xs | null xs = 0
      | otherwise = sum (map realToFrac xs) / fromIntegral (length xs)

_indexString :: String -> Int -> String
_indexString s i =
  let idx = if i < 0 then i + length s else i
  in if idx < 0 || idx >= length s
       then error "index out of range"
       else [s !! idx]

_input :: IO String
_input = getLine


reverseKGroup :: [Int] -> Int -> [Int]
reverseKGroup nums k = fromMaybe ([]) $
    (let n = length nums in case if (k <= 1) then Just (nums) else Nothing of Just v -> Just v; Nothing -> (let result = [] in (let i = 0 in Just (result))))
  where
    n = length nums

main :: IO ()
main = do
    return ()

File: examples/leetcode-out/swift/25/reverse-nodes-in-k-group.swift
import Foundation

func reverseKGroup(_ nums: [Int], _ k: Int) -> [Int] {
	let nums = nums
	let k = k
	
	let n = nums.count
	if k <= 1 {
		return nums
	}
	var result: [Int] = []
	var i = 0
	while i < n {
		let end = i + k
		if end <= n {
			var j = end - 1
			while j >= i {
				result = result + [nums[j]]
				j = j - 1
			}
		} else {
			var j = i
			while j < n {
				result = result + [nums[j]]
				j = j + 1
			}
		}
		i = i + k
	}
	return result
}

func main() {
}
main()
